qiskit/__init__.py

```diff
@@ -14,9 +14,31 @@
 
 """Main Qiskit public functionality."""
 
+import os
 import pkgutil
 import sys
 import warnings
+import importlib.metadata
+
+try:
+    _qiskit_version = importlib.metadata.version("qiskit")
+except importlib.metadata.PackageNotFoundError:
+    # Terra doesn't care if there's no metapackage installed.
+    pass
+else:
+    _major, _ = _qiskit_version.split(".", 1)
+    _suppress_error = os.environ.get("QISKIT_SUPPRESS_1_0_IMPORT_ERROR", False) == "1"
```

Comment:
Do you think this should be documented anywhere? We do have: https://docs.quantum.ibm.com/start/configure-qiskit-local#environment-variables for env variables, but this seems esoteric and dangerous enough that it's probably not worth putting there.

Reply:
It's at the very very bottom of https://qisk.it/packaging-1-0, and in some ways I was deliberately trying to obfuscate it because I don't want people to use it.  It's only intended as a relief valve if I've made a mistake in this PR.

I thought people who know enough to use it safely are likely to check the source code to see what checks it's doing before raising the error, in which case they'll see the envvar.